strawberry/schema/execute.py

```diff
@@ -124,16 +125,28 @@ async def execute(
 
             async with extensions_runner.executing():
                 if not execution_context.result:
-                    result = original_execute(
-                        schema,
-                        execution_context.graphql_document,
-                        root_value=execution_context.root_value,
-                        middleware=extensions_runner.as_middleware_manager(),
-                        variable_values=execution_context.variables,
-                        operation_name=execution_context.operation_name,
-                        context_value=execution_context.context,
-                        execution_context_class=execution_context_class,
-                    )
+                    if execution_context.operation_type == OperationType.SUBSCRIPTION:
+                        # TODO: should we process errors here?
+                        # TODO: make our own wrapper?
+                        return await subscribe(  # type: ignore
+                            schema,
+                            execution_context.graphql_document,
+                            root_value=execution_context.root_value,
+                            context_value=execution_context.context,
+                            variable_values=execution_context.variables,
+                            operation_name=execution_context.operation_name,
+                        )
```

Comment:
This extends the `Schema.execute` method to handle subscriptions. Im wondering whether this was intentional since we already have a separate `Schema.subscribe` method? :eyes: @patrick91 @bellini666 

I noticed this change while upgrading to the latest version. In my codebase mypy now complains everywhere that `Schema.execute` can now return both `ExecutionResult` and `SubscriptionExecutionResult`. Before `Schema.execute` could only return an `ExecutionResult` while subscriptions were handled by `Schema.subscribe`.

Reply:
that was poor design on my part, it will be fixed in #3554 😊